flask_common/mongo/documents.py

```diff
@@ -18,7 +20,7 @@
 
 class StringIdField(StringField):
     def to_mongo(self, value):
-        if not isinstance(value, basestring):
+        if not isinstance(value, (str, bytes)):
```

Comment:
I would advise against using `past.builtins.basestring` for at least two reasons:

1. In 50% of the cases, py3 bytes won't work with code that follows isinstance check as it is. Py3's `bytes` behave significantly different, so you might have to revisit this piece anyway when you actually try to run it on py3. Better expose potential compatibility bugs now than later.
2. When we move to py3 and be confident with it, I'd like to drop support of python 2 to reduce maintenance effort. That would include removing `six`/`future`. If you use `(str, bytes)`, all you need to remove `future` is to remove the import, which can be done with a mostly-mechanical search-and-replace. In case of a `past` import, you'll have to revisit the actual code again.

So please don't make more work for your future self and drop `basestring`.

Reply:
I hadn't noticed that `basestring` matches `bytes` as well. Then let's not keep it.